authorizations/authorization-ranger/src/test/java/org/apache/gravitino/authorization/ranger/integration/test/RangerHiveE2EIT.java

```diff
@@ -204,52 +229,327 @@ public void stop() {
+    catalog.asTableCatalog().dropTable(NameIdentifier.of(schemaName, tableName));
+    catalog.asSchemas().dropSchema(schemaName, true);
+    metalake.deleteRole(readOnlyRole);
+  }
+
+  @Test
+  void testCreateAllPrivilegesRole() throws InterruptedException {
+    String roleName = "allPrivilegesRole";
+    SecurableObject securableObject =
+        SecurableObjects.ofMetalake(
+            metalakeName,
+            Lists.newArrayList(
+                Privileges.CreateCatalog.allow(),
+                Privileges.UseCatalog.allow(),
+                Privileges.UseSchema.allow(),
+                Privileges.CreateSchema.allow(),
+                Privileges.CreateFileset.allow(),
+                Privileges.ReadFileset.allow(),
+                Privileges.WriteFileset.allow(),
+                Privileges.CreateTopic.allow(),
+                Privileges.ConsumeTopic.allow(),
+                Privileges.ProduceTopic.allow(),
+                Privileges.CreateTable.allow(),
+                Privileges.SelectTable.allow(),
+                Privileges.ModifyTable.allow(),
+                Privileges.ManageUsers.allow(),
+                Privileges.ManageGroups.allow(),
+                Privileges.CreateRole.allow()));
+    metalake.createRole(roleName, Collections.emptyMap(), Lists.newArrayList(securableObject));
+
+    // Granted this role to the spark execution user `HADOOP_USER_NAME`
+    String userName1 = System.getenv(HADOOP_USER_NAME);
+    metalake.grantRolesToUser(Lists.newArrayList(roleName), userName1);
+
+    waitForUpdatingPolicies();
+
+    // Test to create a schema
+    sparkSession.sql(SQL_CREATE_SCHEMA);
+
+    // Test to creat a table
+    sparkSession.sql(SQL_USE_SCHEMA);
+    sparkSession.sql(SQL_CREATE_TABLE);
+
+    // Clean up
+    catalog.asTableCatalog().dropTable(NameIdentifier.of(schemaName, tableName));
+    catalog.asSchemas().dropSchema(schemaName, true);
+    metalake.deleteRole(roleName);
+  }
+
+  @Test
+  void testDeleteAndRecreateRole() {
     // Create a role with CREATE_SCHEMA privilege
-    SecurableObject securableObject1 =
+    String roleName = "createSchemaRole";
+    SecurableObject securableObject =
+        SecurableObjects.parse(
+            String.format("%s", catalogName),
+            MetadataObject.Type.CATALOG,
+            Lists.newArrayList(Privileges.UseCatalog.allow(), Privileges.CreateSchema.allow()));
+    metalake.createRole(roleName, Collections.emptyMap(), Lists.newArrayList(securableObject));
+
+    // Granted this role to the spark execution user `HADOOP_USER_NAME`
+    String userName1 = System.getenv(HADOOP_USER_NAME);
+    metalake.grantRolesToUser(Lists.newArrayList(roleName), userName1);
+
+    // Delete the role
+    metalake.deleteRole(roleName);
+
+    // Create the role again
+    metalake.createRole(roleName, Collections.emptyMap(), Lists.newArrayList(securableObject));
+
+    // Grant the role again
+    metalake.grantRolesToUser(Lists.newArrayList(roleName), userName1);
+
```

Comment:
I think it better to add some Spark tests in the here, confirm the role is created successfully again.

Reply:
I added.